src/Nethermind/Nethermind.Synchronization/Peers/SyncPeerAllocation.cs

```diff
@@ -29,14 +29,15 @@ public class SyncPeerAllocation
         public bool HasPeer => Current is not null;
 
         public SyncPeerAllocation(PeerInfo peerInfo, AllocationContexts contexts)
-            : this(new StaticStrategy(peerInfo), contexts)
+            : this(new StaticStrategy(peerInfo), contexts, null)
         {
         }
 
-        public SyncPeerAllocation(IPeerAllocationStrategy peerAllocationStrategy, AllocationContexts contexts)
+        public SyncPeerAllocation(IPeerAllocationStrategy peerAllocationStrategy, AllocationContexts contexts, Lock? allocationLock)
```

Comment:
Make:
```suggestion
        public SyncPeerAllocation(IPeerAllocationStrategy peerAllocationStrategy, AllocationContexts contexts, Lock? allocationLock = null)
```

and don't pass `null` when not needed?